_posts/en/newsletters/2019-05-14-newsletter.md

```diff
@@ -0,0 +1,485 @@
+use cases could be transferred to Taproot.  The best place to start is by
+looking at the way most wealth is transferred via the Bitcoin protocol
+right now: single-sig P2PKH and P2WPKH spends, which constitute FIXME:x%
+of all spends.  FIXME:p2sh.info-ref
+
+Single-sig P2WPKH wallets today currently generate a private key, derive
+a public key (pubkey) from it, and hash that pubkey to create the
+witness program for a bech32 address.  (P2PKH is functionally identical
+but uses a different script and a different address encoding.)
+
+| Object      | Operation                                                       | Example result |
+|-|-|-|
+| Private key | read 32 bytes from [CSPRNG][], or using [BIP32][] HD derivation | 0xFIXME |
+| Public key  | point(0xFIXME), or using [BIP32][] HD public derivation         | 0x02FIXME |
+| Hash        | ripemd(sha256(0x02FIXME))                                       | 0xFIXME |
+| Address     | bech32.encode(0, 0xFIXME)                                       | bc1qFIXME |
+
+Under Taproot, the hashing step will be skipped and so the bech32
+address will contain the pubkey directly, with one small change.
+Currently, 33-byte Bitcoin-style pubkeys are encoded to start with either
+a 0x02 or 0x03 to allow validators to reconstruct the key's Y-coordinate
+on the secp256k1 elliptic curve; in bip-taproot, the value of this byte
+is reduced by two so that 0x02 becomes 0x00 and 0x03 becomes 0x01,
+although the meaning is exactly the same.  Also the witness version is
+changed from the `0` used for P2WPKH/P2WSH to a `1`.
+
+| Object           | Operation                                                       | Example result |
+| Private key      | read 32 bytes from [CSPRNG][], or using [BIP32][] HD derivation | 0xFIXME |
+| Public key       | point(0xFIXME), or using [BIP32][] HD public derivation         | 0x02FIXME |
+| Alter key prefix | (key[0] - 2),key[1:33])                                         | 0x00FIXME |
+| Address          | bech32.encode(1, 0x00FIXME)                                     | bc1pFIXME |
+
+Here's an example of existing addresses compared to an example taproot
+address.
+
+    p2pkh FIXME
+    p2sh FIXME
+    p2wpkh FIXME
+    p2wsh FIXME
+    taproot FIXME
+
+Spending from P2PKH or P2WPKH requires the spender include their public
+key in their input.  For Taproot, the public key was provided in the
+UTXO being spent, so several vbytes are saved by not including it again.
+The spend must also include a signature; this is a Schnorr-style
+signature as defined by [bip-schnorr][] with an optional sighash byte
+appended.  If the default sighash is used, the signature is 64 bytes (16
+vbytes); if a non-default is used, it is 65 bytes (16.25
+vbytes[^vbytes-decimal]).  Overall, this makes the cost to create and
+spend a Taproot single-sig output about 1% more expensive than P2WPKH.
+However, the costs are distributed differently---it costs more to create
```

Comment:
> I have a slight concern that many wallets won't upgrade because "why do the work to pay 1% more fees."

Me too.  However there are huge advantages for multisig users and advanced script users, so they can lead the way.  More importantly, the reduced costs of multisig and advanced scripts will help attract users to more secure methods of using Bitcoin than single-sig, reducing the number of single-sig users.  When enough people are using Taproot anyway, then there's privacy advantage to using it for single-sig too, which may outweigh the 1% difference.

> if we look at common wallet spends such as 2 input, 1 output or 3 input, 2 output, those become meaningfully more economical using Taproot.

Right, but somebody had to pay the fees to create those inputs' prevouts and that probably affected the amount of money you received.  This is like insisting that companies sent you their bills with business return envelopes so you don't have to provide your own stamp; that may seem cheaper to you, but companies are just going to charge you the cost of postage anyway.